aws/iam.go

```diff
@@ -0,0 +1,66 @@
+package aws
+
+import (
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/iam"
+	"github.com/gruntwork-io/cloud-nuke/config"
+	"github.com/gruntwork-io/cloud-nuke/logging"
+	"github.com/gruntwork-io/gruntwork-cli/errors"
+)
+
+// List all IAM users in the AWS account and returns a slice of the UserNames
+// TODO: Implement exclusion by time filter
+func getAllIamUsers(session *session.Session, configObj config.Config) ([]*string, error) {
+	svc := iam.New(session)
+	input := &iam.ListUsersInput{}
+
+	var userNames []*string
+
+	// TODO: Probably use ListUsers together with ListUsersPages in case there are lots of users
+	output, err := svc.ListUsers(input)
+	if err != nil {
+		return nil, errors.WithStackTrace(err)
+	}
+
+	for _, user := range output.Users {
+		if config.ShouldInclude(*user.UserName, configObj.IAMUsers.IncludeRule.NamesRE, configObj.IAMUsers.ExcludeRule.NamesRE) {
```

Comment:
NIT: Use `aws.StringValue` when dereferencing string ptrs from AWS SDK (this is a safe dereference that returns `''` for `nil` instead of `panic`)


```suggestion
		if config.ShouldInclude(aws.StringValue(user.UserName), configObj.IAMUsers.IncludeRule.NamesRE, configObj.IAMUsers.ExcludeRule.NamesRE) {
```